core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowResource.scala

```diff
@@ -101,4 +95,32 @@ class WorkflowResource {
     }
 
   }
+
+  /**
+    * This method deletes the workflow from database
+    *
+    * @param session HttpSession
+    * @return Response, deleted - 200, not deleted - 304 // TODO: change the error code
+    */
+  @DELETE
+  @Path("/{wid}")
+  def deleteWorkflow(@PathParam("wid") wid: UInteger, @Session session: HttpSession): Response = {
+    val user = UserResource.getUser(session)
+    if (user == null) return null
+    if (workflowOfUserFetchOneByID(wid, user.getUid)) {
+      workflowDao.deleteById(wid)
+      Response.ok().build()
+    } else {
+      Response.notModified().build()
+    }
+
+  }
+
+  private def workflowOfUserFetchOneByID(wid: UInteger, uid: UInteger): Boolean = {
```

Comment:
naming doesn't reflect this returns a boolean

Reply:
true. will update.